src/TLuaInterpreter.h

```diff
@@ -433,7 +433,23 @@ class TLuaInterpreter : public QThread
     static int setDiscordCharacterIcon(lua_State* L);
     static int setDiscordCharacter(lua_State* L);
     static int setDiscordArea(lua_State *L);
+    static int setDiscordPresenceId(lua_State*);
+    static int isUsingDefaultDiscordPresenceId(lua_State*);
+    static int setDiscordStateText(lua_State*);
+    static int setDiscordDetailText(lua_State*);
+    static int setDiscordLargeIcon(lua_State*);
+    static int setDiscordLargeIconText(lua_State*);
+    static int setDiscordSmallIcon(lua_State*);
+    static int setDiscordSmallIconText(lua_State*);
+    static int getDiscordStateText(lua_State*);
+    static int getDiscordDetailText(lua_State*);
+    static int getDiscordLargeIcon(lua_State*);
+    static int getDiscordLargeIconText(lua_State*);
+    static int getDiscordSmallIcon(lua_State*);
+    static int getDiscordSmallIconText(lua_State*);
```

Comment:
Why these functions duplicate the API I've already got?

Reply:
The `setXxxx` ones work directly on the Rich Presence entry directly and will work with `any` Presence Id - whereas your *fancy* :smile: ones can **only** work with the Mudlet one (which requires, for instance, that there is an icon with the same name as the MUD and that is what is used for the Large icon) - and now throw an `nil+error message` if that is not the one in use for the given profile.

Reply:
I have some more commits pending that:
+ adds the needed conditional compilation for those cases where Discord is not available (32-bit Linux - e.g. x86 Debian; FreeBSD)
+ MIT license entry in the `About Dialog`
+ fix a bug where I'd forgotten to save the entered Presence Id to the profile
+ fix a bug where the initial opening of the profile when another Presence Id was to be used for that profile was not initializing the remote Discord RP with that Id until the first change was made to any contents.

I have to run them past a Linux platform as well as the Windows one that I was using and check that FreeBSD still works - the last two are okay right now but I hit something late last night and need to resolve it but have some RL stuff I MUST do this morning.

Reply:
We can't have two APIs that duplicate each other, that'll be confusing for users and not really smart on our part.

Reply:
We'll have to merge them somehow - make the existing API work with any presence ID and add the getters.

Reply:
Your existing API **cannot** (IMHO) work for anything but Mudlet's Presence as far as I can tell - because we will not control what "keys" other Guilds will use for their icons or even how they want to use those icons.  For example I am developing something with one MUD that will use a range of icons to show the side of the player and their level as the Large Icon - which is totally different from using a single icon allocated to them based on the MUD name.

Reply:
I don't see why is that an issue - they can use whatever keys they'd like

Reply:
I'll busy today so I can't think on a solution for this, could you think of one?

Reply:
Your solution is predicated - not unreasonably BTW - on a single large icon for each Mud that Mudlet knows about and has entered into the `QMap<Host*, QString>mGamesNames` and which is selected when the lua function `setDiscordGame` is used - however should a MUD want to have a range of icons they cannot do that within your API.  Also reporting `Playing Mudname` in the *detail* text is not useful when the line above that will already say `Mudname` in bold - whilst I have tentatively changed it to say `via Mudlet (connection state)` that is entirely usable to say something else (of course we hope that a Mud Server *Discord* package might at least mention us but they would not have to...)

:bulb: It might be possible to replace your four C++ coded functions to ones in lua that call the lower level ones that I have produced - indeed that would mean I could simplify mine... how would that work with the gmcp and other OOB protocols that you are looking to tie into this?

Reply:
It would work fine, the game would just transmit the Discord ID as well.

I also think that it's more intuitive to use `setDiscordCharacterIcon` as a name than `setDiscordSmallIcon`, it makes more sense no?

Reply:
Only if you are using the Small Icon for that purpose! Me and Austin (an Immortal of the MUD Server + the Discord Guild owner) on WoTMUD are planning to use the Small Icon to show the Clan affiliation of the character - which would be not used for unclanned and those going incognito.  Other MUDs might want to use it to represent what the character is currently doing...

On the whole, for a generic (*raw*) interface I think a neutral (in this case descriptive) term is a better bet. OTOH For a lua script wrapper that handles things for a specific setup and calls the raw functions, like I think we can convert your API work so far to, it would not be unreasonable to call your function that as part of a *cooked* interface. 

"*It would work fine, the game would just transmit the Discord ID as well.*" - I think I left a commented out suggestion for that in the GMCP code - but I do not know the intricacies of those OOB protocols so cannot be certain it is correct.

Reply:
Yeah, you can still use `setDiscordCharacterIcon` for any icon, it's not like Mudlet is going to validate that the image represents a character :rofl: 

Okay, let's drop that API though, we'll also have to change the privacy settings to reflect it. Lets go with small icon / large icon and so on.